src/jira/client/axios.ts

```diff
@@ -55,6 +55,22 @@ function getAuthMiddleware(secret: string) {
 	);
 }
 
+/**
+ * Wrapper for AxiosError, which includes error status
+ */
+export class JiraClientError extends Error {
```

Comment:
Instead of doing a custom error, you can modify the `errorSerializer` in logger.ts to fix this issue globally.  But you'll have to revert the `{ err }` change above.

Reply:
That's the way I've chosen exactly to avoid doing any modifications in errorSerializer. 

Firstly I've implemented with errorSerializer, but that didn't seem clean to me for the following reasons.

We can't detect if it is AxiosError in the serialiser because it is interface, so we can't do instanceof. 
AxiosError implements its own "toJSON" function, which prevents us from adding fields to it (they won't be logged unless we override toJSON there. 

Another reason to use our own exception type is to enrich it with a "status" parameter which we'll further use for error handling

Reply:
We can detect it and we can use `instanceof AxiosError` since the error coming into the serializer is not typed at all.  You can even user `err.name` instead if you wanted to for using a switch statement.

The logger doesn't have to use the `.toJSON` function and it's a non-standard function anyways.  You could override it, but frankly, this is the whole point of the serializer in the first place.

AxiosError already has a code/status value.  No need to add an extra class.

Reply:
right, AxiosError is a type not a class, which is why all axios errors has a boolean field of `isAxiosError` to know that the error is specific to axios.

Reply:
also, added benefit of having this in the global serializer is that we could test it more thoroughly, which wouldn't be a bad idea for this.

Reply:
Having status in the exception is very handy @mboudreau because based on it we could do different strategies for SQS retries. Also, we already have status field in Oktokit's client exception, thus this change would make things more uniform.

Reply:
We just brainstormed it with Tarak and we concluded that:
* It is beneficial to remap AxiosError to our internal error model to have stricter typing. It will allow to separate Axios http client domain model from our App domain model. 
* We'll have more control over the error (Aadding status without adding a mess to existing error object)
* Unification. We'll have to map some of errors in GitHub client to RateLimitingError (like we do now in octokit). So using our own errors everywhere will make the approach more uniformed.

It will significantly simplify error handling